README.md

```diff
@@ -98,14 +103,92 @@ We have a small collection of algorithms, one for every day of the month. Scroll
       left => right
     ```
 
+###  **December 6 - LCM**
+- **Problem**
+  - Return the least common multiple of two or more numbers.
+  - Use the `greatest common divisor (GCD)` formula and the fact that `lcm(x,y) = x * y / gcd(x,y)` to determine the least common multiple.
+  - **Hint:** The GCD formula might use recursion.
+- **Sample Run**
+  ```bash
+  lcm(12, 7) # 84
+  lcm([1, 3, 4], 5) # 60
+  ```
+
+###  **December 7 - Are they approximately equal?**
+- **Problem**
+  - Checks if two numbers are approximately equal to each other.
+  - `Optional:` Add Tolerance level as a third parameter
+- **Sample Run**
+  - Without Tolerance Level:
+  ```bash
+  IsApproximatelyEqual(3.0, 2.5706); // true
+  ```
+  - With Tolerance Level:
+  ```bash
+  IsApproximatelyEqual(3.0, 2.5706, 0.01); // false as the difference is more than the Tolerance Level
+  ```
+
+###  **December 8 - Singular-Plural**
+- **Problem**
+  - Returns the singular or plural form of any regular nouns based on the input number.
+  - If `num` is either `-1 or 1`, return the singular form of the word. If `num` is any other number, return the plural form.
+  - Also if both the arguments are strings, output the singular and plural word from the input given.
+  - `Optional:` To spice things up, use a dictionary file to compare give the appropriate outputs. Therefore this will cover Irregular nouns in it too.
+- **Sample Run**
+  ```bash
+  > SingularPlural("Apple", 2)
+      Apples
+  > SingularPlural("Apple", 1)
+      Apple
+  > SingularPlural("Apples", 1)
+      Apple
+  > SingularPlural("Apples", "Apple")
+      Singular: Apple
+      Plural: Apples
+  > SingularPlural("Apples", "Orange")
+      Invalid Input
+  ```
+- **Resources**
+  - [Plural Nouns: Rules and Examples](https://www.grammarly.com/blog/plural-nouns/)
+
+###  **December 9 - Is this an URL?**
+- **Problem**
+  - Returns `true` if the given string is an absolute URL, `false` otherwise.
+  - **Hint:** Regular Expressions can be used to solve this problem.
+- **Sample Run**
+  ```bash
+  > IsURL("https://duckduckgo.com/?q=test+string&atb=v143-2__&ia=web")
+    True
+  > IsURL("Martha?")
+    False
+  > IsURL("https://duckduckgo")
+    False
+  ```
+- **Resources**
+  - [Regular Expressions in C++](http://www.cplusplus.com/reference/regex/)
+  - [Regular Expressions in Python](https://www.tutorialspoint.com/python/python_reg_expressions.htm)
+  - [Regular Expressions in Java](https://www.javatpoint.com/java-regex)
+
+###  **December 10 - Find the Determinant**
```

Comment:
Add a resources section for what a determinant is.
https://www.mathsisfun.com/algebra/matrix-determinant.html